src/redist/targets/GenerateLayout.targets

```diff
@@ -7,14 +7,19 @@
 
   <PropertyGroup>
     <!-- Blob storage directories are not stabilized, so these must refer to a package that does not stabilize -->
-    <AspNetCoreBlobVersion>$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreBlobVersion>
-    <CoreSetupBlobVersion>$(MicrosoftNETCoreAppRuntimePackageVersion)</CoreSetupBlobVersion>
+    <!-- In source build, the layout does match, so use the runtime package versions rather than the VS redist versions -->
+    <AspNetCoreBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreBlobVersion>
+    <AspNetCoreBlobVersion Condition=" '$(DotNetBuildFromSource)' != 'true' ">$(VSRedistCommonAspNetCoreSharedFrameworkx6470PackageVersion)</AspNetCoreBlobVersion>
+    <CoreSetupBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftNETCoreAppRuntimePackageVersion)</CoreSetupBlobVersion>
```

Comment:
I think this may also need `AND '$(DotNetBuildOffline)' == 'true'`

Reply:
The way `DotNetBuildOffline` has been included in 6.0, it is always true in a tarball build and false in repo's source-build CI legs.  I'm not sure what these versions should be in the source-build leg.